### PR TITLE
[ExportVerilog] Fix an iterator update bug

### PR DESCRIPTION
**Pull request by uenoku**
This fixes a bug introduced by 286c4830125. An iterator is mistakenly incremented in certain
execution. Since `opIterator` is incremented in the beginning of the loop, 
we are not actually processing next operations. This causes an assertion failure
on ExportVerilog in internal designs. 

`opIterator` is incremented at:
https://github.com/llvm/circt/blob/a9a30f5ecdb215b82dee6c422ed87c872bb3a0bd/lib/Conversion/ExportVerilog/PrepareForEmission.cpp#L392-L395

### CODE REVIEW

**Review by fabianschuiki — APPROVED** [review]
Nice catch! :tada:

**Comment by uenoku** [comment]
It seems my fix might be wrong, I'll make it this draft until it gets clear.

**Comment by uenoku** [comment]
```mlir
module attributes {circt.loweringOptions = "maximumNumberOfTermsPerExpression=2"} {
  hw.module @Foo(%a: i1) {
    sv.always posedge %a {
      %0 = comb.and %a, %a, %a, %a : i1
      %1 = comb.and %a, %a, %a, %a : i1
      sv.if %0 {
        sv.error "bar"
      }
      sv.if %1 {
        sv.error "foo"
      }
    }
    hw.output
  }
}
```

File: test/Conversion/ExportVerilog/max-terms.mlir [hunk 1]
```diff
@@ -36,3 +36,19 @@ hw.module @large_use_in_procedural(%clock: i1, %a: i1) {
...
+    // CHECK:      assign {{.*}} = a & a & a & a & a;
+    // CHECK-NEXT: assign {{.*}} = a & a & a & a & a;
+    sv.if %0 {
+      sv.exit
+    }
+    sv.if %1 {
+      sv.exit
+    }
+  }
+}
```
Comment (mikeurbach) [inline test/Conversion/ExportVerilog/max-terms.mlir:54]: super nit, but github points out this file doesn't end in a newline. it's not required llvm style, but maybe it's worth having?

**Review by mikeurbach — APPROVED** [review]
Thanks, looks good to me. I believe this is also covered by the "short integration tests" premerge check, which is great.

File: test/Conversion/ExportVerilog/max-terms.mlir (hunk 1 above)
Comment (uenoku) [reply test/Conversion/ExportVerilog/max-terms.mlir:54]: Yes, thanks!

Comment (uenoku) [reply test/Conversion/ExportVerilog/max-terms.mlir:54]: Yes, thanks!

**Pull request merged**